python/uds.py

```diff
@@ -641,17 +641,17 @@ def access_timing_parameter(self, timing_parameter_type: TIMING_PARAMETER_TYPE,
       parameter_values = resp
       return parameter_values
 
-  def secured_data_transmission(self, data: bytes):
+  def secured_data_transmission(self, data: bytes) -> bytes:
     # TODO: split data into multiple input parameters?
     resp = self._uds_request(SERVICE_TYPE.SECURED_DATA_TRANSMISSION, subfunction=None, data=data)
     # TODO: parse response into multiple output values?
     return resp
 
-  def control_dtc_setting(self, dtc_setting_type: DTC_SETTING_TYPE):
+  def control_dtc_setting(self, dtc_setting_type: DTC_SETTING_TYPE) -> None:
     self._uds_request(SERVICE_TYPE.CONTROL_DTC_SETTING, subfunction=dtc_setting_type)
 
   def response_on_event(self, response_event_type: RESPONSE_EVENT_TYPE, store_event: bool, window_time: int,
-                        event_type_record: int, service_response_record: int):
+                        event_type_record: int, service_response_record: int) -> Mapping[str, Any]:
```

Comment:
this might actually be a good usecase for a [TypedDict](https://peps.python.org/pep-0589) to make the return type less generic

Reply:
would it be okay to return `"event_window_time": None` in the case `response_event_type == RESPONSE_EVENT_TYPE.REPORT_ACTIVATED_EVENTS` below? that way, the TypedDict can preserve [totality](https://peps.python.org/pep-0589/#totality)

Reply:
let me know if you'd like me to implement either